src/pool/srv_pool.c

```diff
@@ -4586,6 +4586,7 @@ ds_pool_update_handler(crt_rpc_t *rpc)
 		DP_UUID(in->pti_op.pi_uuid), rpc, DP_RC(rc));
 	crt_reply_send(rpc);
 	pool_target_addr_list_free(&inval_list_out);
+	pool_target_addr_list_free(&list);
```

Comment:
The list is from rpc, I thought rpc destory should take care of this?

Reply:
No. Unfortunately this is in reply of RPC and user should clear it manually. Mercury free only structs previously decoded (in structures).

Reply:
hmm, in->pti_addr_list.ca_arrays is allocated by HG when decoding the RPC request, and also will be freed by HG when calling HG_Free_output (called by cart). 
So should need not explicitly free it by daos code here, TBH, I will be very surprise if it is not working as that.

Reply:
Hmm. It looks - yes. This is my fault. It's come from in_struct and will be freed automatically. Removed.